zerberus/zone.cpp

```diff
@@ -44,18 +44,22 @@ Zone::~Zone()
 
 bool Zone::match(Channel* c, int k, int v, Trigger et, double rand)
       {
-      int cc64 = c->sustain();
 
       if ((k >= keyLo)
          && (k <= keyHi)
          && (v >= veloLo)
          && (v <= veloHi)
          && (loRand <= rand && hiRand > rand)
          && (et == trigger)
-         && (cc64 >= locc[64] && cc64 <= hicc[64])
          ) {
 //printf("   Zone match %d %d %d -- %d %d  %d %d  center %d trigger %d\n",
 //         k, v, et, keyLo, keyHi, veloLo, veloHi, keyBase, trigger);
+            for (int i = 0; i < 128; i++) {
```

Comment:
Optimisation: we could store at the voice level a bool to indicate that we have set one locc or hicc. It would avoid to loop through 128 "empty" locc/hicc.